src/irmin-pack/checks.ml

```diff
@@ -363,3 +421,167 @@ module Index (Index : Pack_index.S) = struct
     Utils.Progress.finalise bar;
     result
 end
+
+module Stats (S : sig
+  type step
+
+  val step_t : step Irmin.Type.t
+
+  module Hash : Irmin.Hash.S
+end) =
+struct
+  type path = S.step option list
```

Comment:
Perhaps worth adding a comment to say that `None` corresponds to Inodes, or else using `type step = Node of S.step | Inode` for clarity.